proxy.go

```diff
@@ -232,19 +190,39 @@ func executeWithRetry(
 		}
 		// The request has been successfully proxied.
 
-		since = time.Since(startTime).Seconds()
-		monitorDuration(since)
-
 		srw.SetStatusCode(rw.StatusCode())
 		// StatusBadGateway response is returned by http.ReverseProxy when
 		// it cannot establish connection to remote host.
 		if rw.StatusCode() == http.StatusBadGateway {
 			log.Debugf("the invalid host is: %s", s.host.addr)
 			s.host.penalize()
-			q := getQuerySnippet(req)
-			err1 := fmt.Errorf("%s: cannot reach %s; query: %q", s, s.host.addr.Host, q)
-			respondWith(srw, err1, srw.StatusCode())
+			s.host.dec()
+			if numRetry < maxRetry && s.host.replica.cluster.getHost().isActive() {
```

Comment:
cf @Garnek20 If the scope contains a sessionID the retry shouldn't be done because user are expecting to run a query an a specfic server (look at how the getHostSticky is used to understand what I mean)

Reply:
Indeed. @JingmaoYou you can assess that with:
```
s.sessionId != ""
```

or even better if you added a method to scope:
```
func (s * scope) StickModyEnabled() bool
```

Reply:
also @JingmaoYou 
```
s.host.replica.cluster.getHost()
```
seems to perform a side effect, therefore you should not call it twice (I see it also in line 208).
It's not good that this method does any side effect, but let's not change that with this feature

Reply:
The other observation here, I think we need to make sure the query time is below the maxExecutionTime, including the time for retry.

Reply:
For the function StickModyEnabled(), it is just used to judge if the sessionId of the scope is "" ? @Garnek20

Reply:
`The other observation here, I think we need to make sure the query time is below the maxExecutionTime, including the time for retry.` 
Good point.
FYI, whne I reviewed this PR, I thought it was already the case seems the context you use on each retry is the same but didn't look deeper to see if it was really the case.

Reply:
Hello, I have checked the code, I use the same context during the retry. We will do the retry if query has no error(deadline error for example). If there is a deadline error, it will return the error. 
From what I could see, there is no issue here.